azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -60,8 +64,20 @@ def _create_mqtt_client(self):
         def on_connect(client, userdata, flags, rc):
             logger.info("connected with result code: {}".format(rc))
 
-            # TODO: how to do failed connection?
-            if self.on_mqtt_connected:
+            if rc:
```

Comment:
>if rc [](start = 12, length = 5)

RC always indicates error? There's no success RC?

Reply:
Yes, rc==0 is success.

---
In reply to: [303131872](https://github.com/Azure/azure-iot-sdk-python-preview/pull/136#discussion_r303131872) [](ancestors = 303131872)

Reply:
I think a brief comment to that effect could be helpful, i.e. ```if rc:    #rc 0 == success```

---
In reply to: [305849146](https://github.com/Azure/azure-iot-sdk-python-preview/pull/136#discussion_r305849146) [](ancestors = 305849146,303131872)